packages/website/src/docs/dialog.mdx

```diff
@@ -1,6 +1,146 @@
+  );
+};
+```
+
+### children
+
+<Prop>ReactNode</Prop>
+
+The content to render inside the Dialog.
+
+### isOpen
+
+<Prop>Boolean</Prop>
+
+Controls whether the Dialog is open or not.
+
+### canDismiss
+
+<Prop isOptional default="true">
+  Boolean
+</Prop>
+
+Whether the dialog can be dismissed by clicking outside or pressing the escape key.
+
+```js live removeFragment
+() => {
+  const [isOpen, setIsOpen] = React.useState(false);
+
+  return (
+    <>
+      <Button onClick={() => setIsOpen(true)}>Open dialog</Button>
+      <Dialog isOpen={isOpen} canDismiss={false}>
+        <Box p={6}>
+          <Text as="p" mb={3}>
+            This dialog cannot be dismissed by clicking outside or pressing the
+            escape key.
+          </Text>
+          <Button onClick={() => setIsOpen(false)}>
+            Click to close dialog
+          </Button>
+        </Box>
+      </Dialog>
+    </>
+  );
+};
+```
+
+### onDismiss
+
+<Prop isOptional>Function</Prop>
+
+A function that gets called when the Dialog is dismissed via click outside or escape key.
+
+```js live removeFragment
+() => {
+  const [isOpen, setIsOpen] = React.useState(false);
+
+  return (
+    <>
+      <Button onClick={() => setIsOpen(true)}>Open dialog</Button>
+      <Dialog
+        isOpen={isOpen}
+        onDismiss={() => {
+          alert('Dialog dismissed');
+          setIsOpen(false);
+        }}
+      >
+        <Box p={6}>
+          <Text as="p">
+            Dismissing the dialog by clicking outside or by pressing escape will
+            trigger the `onDismiss` callback.
+          </Text>
+        </Box>
+      </Dialog>
+    </>
+  );
+};
+```
+
+### refToFocusOnOpen
+
+<Prop isOptional>RefObject</Prop>
+
+A reference to an element to focus on inside the dialog after it opens. If not provided, it will focus on the first focusable element.
+
+```js live removeFragment
+() => {
+  const [isOpen, setIsOpen] = React.useState(false);
+  const inputRef = React.useRef(null);
+
+  return (
+    <>
+      <Button onClick={() => setIsOpen(true)}>Open dialog</Button>
+
+      <Dialog
+        isOpen={isOpen}
+        onDismiss={() => setIsOpen(false)}
+        refToFocusOnOpen={inputRef}
+      >
+        <Box p={6} position="relative">
+          <Box position="absolute" top={1} right={1}>
+            <GhostButton size={1} onClick={() => setIsOpen(false)}>
+              <CrossIcon />
+            </GhostButton>
+          </Box>
+          <Heading size={0} fontWeight={500}>
+            Confirm password
+          </Heading>
+          <Input ref={inputRef} type="password" placeholder="******" my={4} />
+          <Flex justifyContent="flex-end">
+            <Button mr={3} onClick={() => setIsOpen(false)}>
+              Cancel
+            </Button>
+            <Button variant="blue" onClick={() => setIsOpen(false)}>
+              Confirm
+            </Button>
+          </Flex>
+        </Box>
+      </Dialog>
+    </>
+  );
+};
+```
+
+### refToFocusOnClose
```

Comment:
shall we add an example for this prop?

Reply:
Probably, but I couldn't thing of one that made sense.